### PR TITLE
Add requirement for Custom Resources to VPA FAQ

### PR DESCRIPTION
**Pull request by voelzmo**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:
As discussed in today's SIG meeting: Let's add the contract for VPA's support for Custom Resources more explicitly to the FAQ, as this question [keeps](https://github.com/kubernetes/autoscaler/issues/5836) [coming](https://github.com/kubernetes/autoscaler/issues/5925) [back](https://github.com/kubernetes/autoscaler/issues/3766).

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: vertical-pod-autoscaler/FAQ.md [hunk 1]
```diff
@@ -107,6 +108,13 @@ is serving.
 
 Note: the commands will differ if you deploy VPA in a different namespace.
 
+### How can I apply VPA to my Custom Resource?
+
+The VPA cannot only scale the built-in resources like Deployment or StatefulSet, but also Custom Resources which manage Pods.
+Just like the Horizontal Pod Autoscaler, the [VPA requires that the Custom Resource implements the `/scale` subresource with the optional field `labelSelector`](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource),
+which corresponds to `.scale.status.selector`. VPA doesn't use the `/scale` subresource for the actual scaling, but uses this label selector to identify the Pods managed by a
+Custom Resource. As VPA relies on Pod eviction to apply new resource recommendations, this ensures that all Pods with a matching VPA object are managed by a controller that will recreate them after eviction. Furthermore, it avoids misconfigurations that happened in the past when label selectors were specified manually.
```
Comment (Shubham82) [inline vertical-pod-autoscaler/FAQ.md:116]: nit: Wrap the line
```suggestion
Custom Resource. As VPA relies on Pod eviction to apply new resource recommendations, this
ensures that all Pods with a matching VPA object are managed by a controller that will
recreate them after eviction. Furthermore, it avoids misconfigurations that happened in the
past when label selectors were specified manually.
```

**Review by Shubham82 — CHANGES_REQUESTED** [review]

**Comment by Shubham82** [comment]
@voelzmo content looks good to me, just small nits.

File: vertical-pod-autoscaler/FAQ.md [hunk 2]
```diff
@@ -107,6 +108,13 @@ is serving.
 
 Note: the commands will differ if you deploy VPA in a different namespace.
 
+### How can I apply VPA to my Custom Resource?
+
+The VPA cannot only scale the built-in resources like Deployment or StatefulSet, but also Custom Resources which manage Pods.
+Just like the Horizontal Pod Autoscaler, the [VPA requires that the Custom Resource implements the `/scale` subresource with the optional field `labelSelector`](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource),
+which corresponds to `.scale.status.selector`. VPA doesn't use the `/scale` subresource for the actual scaling, but uses this label selector to identify the Pods managed by a
```
[the remark this answers is not in the document]
Comment (voelzmo) [reply vertical-pod-autoscaler/FAQ.md]: done.

File: vertical-pod-autoscaler/FAQ.md (hunk 1 above)
Comment (voelzmo) [reply vertical-pod-autoscaler/FAQ.md]: done.

**Comment by Shubham82** [comment]
Thanks @voelzmo 
/lgtm

File: vertical-pod-autoscaler/FAQ.md [hunk 3]
```diff
@@ -107,6 +108,17 @@ is serving.
 
 Note: the commands will differ if you deploy VPA in a different namespace.
 
+### How can I apply VPA to my Custom Resource?
+
+The VPA cannot only scale the built-in resources like Deployment or StatefulSet, but also Custom Resources which manage
```
Comment (jbartosik) [inline vertical-pod-autoscaler/FAQ.md:113]:
```suggestion
The VPA can scale not only the built-in resources like Deployment or StatefulSet, but also Custom Resources which manage
```

File: vertical-pod-autoscaler/FAQ.md [hunk 4]
```diff
@@ -107,6 +108,17 @@ is serving.
...
+Pods. Just like the Horizontal Pod Autoscaler, the [VPA requires that the Custom Resource implements the `/scale`
+subresource with the optional field `labelSelector`](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource),
```
Comment (jbartosik) [inline vertical-pod-autoscaler/FAQ.md:115]:
```suggestion
Pods. Just like the Horizontal Pod Autoscaler, the VPA requires that the Custom Resource implements the
[`/scale` subresource](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource)
with the optional field `labelSelector`,
```

**Review by jbartosik — COMMENTED** [review]
Mostly looks good,  I made a couple small suggestions

File: vertical-pod-autoscaler/FAQ.md (hunk 4 above)
Comment (voelzmo) [reply vertical-pod-autoscaler/FAQ.md]: Thanks!

File: vertical-pod-autoscaler/FAQ.md (hunk 3 above)
Comment (voelzmo) [reply vertical-pod-autoscaler/FAQ.md]: Thanks!

**Comment by Shubham82** [comment]
It looks good to me.
/lgtm

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

**Pull request merged**